MidasCivil_Adapter/CRUD/Create/Loads/BarVaryingDistributedLoads.cs

```diff
@@ -92,19 +117,21 @@ private bool CreateCollection(IEnumerable<BarVaryingDistributedLoad> barVaryingD
                             load.ForceAtStart = CreateSingleComponentVector(i, startLoadVectors[i]);
                             load.ForceAtEnd = CreateSingleComponentVector(i, endLoadVectors[i]);
 
-                            foreach (string assignedBar in assignedBars)
+                            for (int j = 0; j < assignedBars.Count; j++)
                             {
-                                midasBarLoads.Add(Adapters.MidasCivil.Convert.FromBarVaryingDistributedLoad(load, assignedBar, "Force", m_forceUnit, m_lengthUnit));
+                                midasBarLoads.Add(Adapters.MidasCivil.Convert.FromBarVaryingDistributedLoad(load, 
+                                    assignedBars[j], "Force", m_forceUnit, m_lengthUnit, startPosition[j], endPosition[j]));
                             }
                         }
                         else
                         {
                             load.MomentAtStart = CreateSingleComponentVector(i - 3, startLoadVectors[i]);
                             load.MomentAtEnd = CreateSingleComponentVector(i - 3, endLoadVectors[i]);
 
-                            foreach (string assignedBar in assignedBars)
+                            for (int j = 0; j < assignedBars.Count; j++)
                             {
-                                midasBarLoads.Add(Adapters.MidasCivil.Convert.FromBarVaryingDistributedLoad(load, assignedBar, "Moment", m_forceUnit, m_lengthUnit));
+                                midasBarLoads.Add(Adapters.MidasCivil.Convert.FromBarVaryingDistributedLoad(load,
+                                    assignedBars[j], "Moment", m_forceUnit, m_lengthUnit, startPosition[j], endPosition[j]));
```

Comment:
Is this going to work with different element lengths? The names will be the same.

Reply:
The different names are not needed because the `DependencyTypes` are dealt with before this.